### PR TITLE
Rating feature for bytes

### PR DESCRIPTION
**Pull request by MSamiTariq**
# Pull Request Template

Refers: https://github.com/RobinNagpal/dodao-ui/issues/178

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [x] I have carefully reviewed my code and believe it is final from my side.
- [x] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [x] I have tested the new feature or updates in both light and dark themes.
- [x] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [x] I have tested my changes on mobile screen sizes.
- [x] I have tested my changes on tablet screen sizes.
- [x] I have tested my changes on laptop screen sizes.

### Functional Testing
- [x] I have thoroughly tested the functionality of my changes.
- [x] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

Tested the ratings feature by giving the tidbits rating and verified the ratings on the `/tidbits/ratings/` page.

### CODE REVIEW

File: src/app/tidbits/rating/[byteId]/page.tsx
Comment (RobinNagpal) [inline src/app/tidbits/rating/[byteId]/page.tsx:1]: We can remove this page

File: src/app/tidbits/view/[...byteIdAndStep]/ByteView.module.scss [hunk 1]
```diff
@@ -8,4 +8,4 @@
     min-width: 767px;
     max-width: 767px;
   }
-}
+    }
```
Comment (RobinNagpal) [inline src/app/tidbits/view/[...byteIdAndStep]/ByteView.module.scss:11]: Fix the formatting

File: src/components/app/Modal/Byte/ByteRatingModal.module.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
+.ModalContent{
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+}
+
+.FeedbackOptionDiv{
+    :hover {
```
Comment (RobinNagpal) [inline src/components/app/Modal/Byte/ByteRatingModal.module.scss:8]: Check formatting. May be we should include sass formatting in the repo itself.  IT can be done with some prettier plugin

Comment (RobinNagpal) [reply src/components/app/Modal/Byte/ByteRatingModal.module.scss:8]: The prettier work can be done later

File: src/components/app/Modal/Byte/ByteRatingModal.module.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
+.ModalContent{
+    display: flex;
```
Comment (RobinNagpal) [inline src/components/app/Modal/Byte/ByteRatingModal.module.scss:2]: try to use tailwind classed for this.

We should have only colors here. 

This is for consistency

File: src/components/app/Modal/Byte/ByteRatingModal.tsx
Comment (RobinNagpal) [inline src/components/app/Modal/Byte/ByteRatingModal.tsx:1]: Use the same common component and refactor that a bit

File: src/components/bytes/Rating/ByteRatingsTable.module.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.agGridWrapper {
+    width: 100%;
+}
```
Comment (RobinNagpal) [inline src/components/bytes/Rating/ByteRatingsTable.module.scss:3]: Check formatting of this

File: src/components/bytes/Rating/ByteRatingsTable.tsx [hunk 5]
```diff
@@ -0,0 +1,218 @@
+'use client';
+import { Grid2Cols } from '@/components/core/grids/Grid2Cols';
+import SpinnerWithText from '@/components/core/loaders/SpinnerWithText';
+import {
+  ByteRating,
+  SpaceWithIntegrationsFragment,
+  useConsolidatedByteRatingQuery,
+  useByteRatingsQuery,
```
Comment (RobinNagpal) [inline src/components/bytes/Rating/ByteRatingsTable.tsx:8]: Recuse common component

File: src/components/bytes/View/RatingByteView.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import ByteRatingsTable from '@/components/bytes/Rating/ByteRatingsTable';
+import PageWrapper from '@/components/core/page/PageWrapper';
+import { SpaceWithIntegrationsFragment } from '@/graphql/generated/generated-types';
+import Link from 'next/link';
+import React from 'react';
+
+export default function ByteRatingView(props: { space: SpaceWithIntegrationsFragment; byteId: string }) {
+  return (
+    <PageWrapper>
+      <div tw="px-4 md:px-0 overflow-hidden">
```
Comment (RobinNagpal) [inline src/components/bytes/View/RatingByteView.tsx:10]: The back button is not needed
